include/swift/AST/DiagnosticsSema.def

```diff
@@ -2805,6 +2805,11 @@ ERROR(assigning_noescape_to_escaping,none,
 ERROR(general_noescape_to_escaping,none,
       "using non-escaping parameter %0 in a context expecting an @escaping closure",
       (Identifier))
+ERROR(assigning_noescape_to_generic_param,none,
+      "assigning non-escaping value to %0 may allow it to escape",
```

Comment:
We're not assigning when we emit the diagnostic -- there's no assignment expression. I think "converting" still makes sense

Reply:
Sure, thanks! Do you think it's worth adding "generic param" before `%0` as well?

Reply:
@slavapestov ^

Reply:
Done! Changed `assigning` to `converting`.